File: CodingTest/LINE/2.py
from itertools import combinations

def solution(sentences, n):
    answer = -1
    alpha = []
    
    #shift O
    for i in sentences:
        for j in i:
            if j != " ":
                if ord('A') <= ord(j) and ord(j) <= ord('Z'):
                    alpha.append(chr(ord(j) + 32))
                else:
                    alpha.append(j)
        alpha = list(set(alpha))

    combo = list(combinations(alpha, n - 1))
    
    max_cnt1 = 0
    for i in combo:
        cnt = 0
        for j in sentences:
            tmp_cnt = 0
            flag = True
            for k in j:
                if k != " ":
                    if k not in i and chr(ord(k) + 32) not in i:
                        flag = False
                        break
                if ord('A') <= ord(k) and ord(k) <= ord('Z'):
                    tmp_cnt += 1
            if flag:
                tmp_cnt += len(j)
            else:
                tmp_cnt = 0
            cnt += tmp_cnt
        if cnt > max_cnt1:
            max_cnt1 = cnt


    for i in sentences:
        for j in i:
            if ord('a') <= ord(j) and ord(j) <= ord('z'):
                alpha.append(j)
        alpha = list(set(alpha))

    combo = list(combinations(alpha, n))

    max_cnt2 = 0
    for i in combo:
        cnt = 0
        for j in sentences:
            tmp_cnt = 0
            flag = True
            for k in j:
                if k != " " and k not in i:
                    flag = False
                    break
            if flag:
                tmp_cnt += len(j)
            else:
                tmp_cnt = 0
            cnt += tmp_cnt
        if cnt > max_cnt2:
            max_cnt2 = cnt

    answer = max(max_cnt1, max_cnt2)
    
    return answer

File: CodingTest/SKHynix/2.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(int n) {
    vector<int> answer;
    vector<vector<int>> v;
    for(int i = 0; i < 4 * (n - 1) + 1; i++){
        vector<int> v2;
        for(int j = 0; j <= 2 * (n - 1); j++){
            v2.push_back(0);
        }
        v.push_back(v2);
    }

    int i = 0;
    int j = n - 1;
    int num = 1;
    v[i][j] = num;

    while(n - 1 > 0){
        for(int k = 0; k < n - 1; k++){
            i++;
            j++;
            num++;
            v[i][j] = num;
        }
        for(int k = 0; k < n - 1; k++){
            i += 2;
            num++;
            v[i][j] = num;
        }
        for(int k = 0; k < n - 1; k++){
            i++;
            j--;
            num++;
            v[i][j] = num;
        }
        for(int k = 0; k < n - 1; k++){
            i--;
            j--;
            num++;
            v[i][j] = num;
        }
        for(int k = 0; k < n - 1; k++){
            i -= 2;
            num++;
            v[i][j] = num;
        }
        for(int k = 0; k < n - 2; k++){
            i--;
            j++;
            num++;
            v[i][j] = num;
        }
        i++;
        j++;
        n--;
        num++;
        v[i][j] = num;
    }

    for(int i = 0; i < v.size(); i++){
        for(int j = 0; j < v[i].size(); j++){
            if(v[i][j] != 0)
                answer.push_back(v[i][j]);
        }
    }

    return answer;
}
